### PR TITLE
Fix pre-commit hook

### PR DESCRIPTION
**Pull request by berndkaiser**
Use sane Set Builtin flags and do not use `$BASH_SOURCE`

## Description

Adjust `.husky/pre-commit` to work with all posix compatible shells.

## Changelog

* `.husky/pre-commit`

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,7 +1,13 @@
-#!/bin/sh
-SCRIPT_DIRECTORY=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+set -eu
+
+SCRIPT_DIRECTORY="$(dirname -- "$( readlink -f -- "$0"; )";)"
+
+echo "$SCRIPT_DIRECTORY"
```
Comment (migerh) [inline .husky/pre-commit:8]: Do we need this or is this a debug/analysis remnant? ;)

Comment (berndkaiser) [reply .husky/pre-commit]: Thx! fixed

**Comment by migerh** [comment]
Checked this in dash & bash on Linux and MacOS and works for me. One minor convenience thing, because I just ran into this issue: If `./husky/_/husky.sh` does not exist, output a message that hints to running `yarn install` first and `exit 1`?

**Review by migerh — APPROVED** [review]

**Comment by berndkaiser** [comment]
> Checked this in dash & bash on Linux and MacOS and works for me. One minor convenience thing, because I just ran into this issue: If `./husky/_/husky.sh` does not exist, output a message that hints to running `yarn install` first and `exit 1`?
> 
> Apart from that: LGTM

I added a file exists check and a small help message

**Pull request merged**